app/javascript/react/components/NewMovieForm.js

```diff
@@ -0,0 +1,56 @@
+import React, {useState} from "react"
+import NewMovieFormTile from "./NewMovieFormTile"
+import { Redirect } from "react-router-dom"
+import ApiErrorList from "./ApiErrorList"
+
+const NewMovieForm = (props) => {
+  const [movie, setMovie] = useState({})
+  const [errors, setErrors] = useState([])
+
+  const addMovie = async (formPayload) => {
+    try {
+      const response = await fetch("/api/v1/movies", {
+        credentials: "same-origin",
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+          "Accept": "application/json"
+        },
+        body: JSON.stringify(formPayload)
+      })
+      if(!response.ok){
+        const errorMessage = `${response.status} (${response.statusText})`
+        const error = new Error(errorMessage)
+        throw(error)
+      }
+      const responseBody = await response.json()
+      if(responseBody.movie){
+        setMovie(responseBody.movie) 
+      } else {
+        setErrors(responseBody.errors)
+      }
+
+    } catch(error) {
+      console.log(`Error in fetch: ${error.message}`)
+    }
+  }
+
+  let fetchReturn
+  if(movie.id) {
+    fetchReturn = <Redirect to={`/movies/${movie.id}`}/>
+  } 
+  if (errors.length > 0) {
+    fetchReturn = <ApiErrorList errors={errors}/>
+  }
```

Comment:
this does not need to be a conditional. `errors` will either be an empty array based on initial state, or will have errors. On line 48 in the returned JSX we would just need to call on this component for your errors - if there are no errors because it's an empty array then nothing will display on the page. But if there are errors, then it will display them

```js
return (
  <div>
      <ApiErrorList errors={errors} />
        ...
  ```